vnext/template/shared-app/src/MainPage.xaml

```diff
@@ -15,7 +15,7 @@
 {{^languageIsCpp}}
         x:Name="reactRootView"
 {{/languageIsCpp}}
-        ComponentName="{{ name }}"
+        ComponentName="{{ mainComponentName }}"
```

Comment:
For my own understanding, how does mismatched component name connect to the metro issue? I would have guessed that we would see failures funding the component from the AppRegistry instead of bundling errors.

Reply:
From the metro callstack, I wonder if the issue is in two parts. E.g.

1. We redbox bc of mismatch
2. Metro barfs on redbox

Where this would fix 1, preventing 2, but there's still am underlying issue.

Reply:
Metro has a bug where it misreports the error message within metro.

The red box on my machine *does* give an error about not finding the component name in the App registry.

Reply:
Good to know. Could we leave the Metro bug open to track it separately?